meson_options.txt

```diff
@@ -0,0 +1,219 @@
+        '-L/usr/lib',
+        '-L/lib'],
+    description : 'Link flags')
+
+
+option(
+    'cxxflags',
+    type : 'array',
+    value : [
+        '-Wno-pedantic',
+        '-Wno-non-virtual-dtor',
+        '-Wno-unused-parameter',
+        '-Wno-variadic-macros',
+        '-Wdeprecated-declarations',
+        '-Wno-missing-field-initializers'],
+    description : 'C build flags')
+
+
+
+
+# link args
+#============================================================================
+
+option(
+    'archive_link_args',
+    type : 'array',
+    value : ['-larchive'],
+    description : 'link args for libbz2')
+
+option(
+    'boost_link_args', 
+    type : 'array', 
+    value : [
+        '-L/usr/lib64',
+        '-lboost_system',
+        '-lboost_context',
+        '-lboost_thread'], 
+    description : 'link args for boost')
+
+option(
+    'brotli_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lbrotlienc',
+        '-lbrotlidec'],
+    description : 'link args for libbrotli')
+
+option(
+    'bz2_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lbz2'],
+    description : 'link args for libbz2')
+
+option(
+    'curl_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lcurl'],
+    description : 'link args for libcurl')
+
+option(
+    'dl_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-ldl'],
+    description : 'link args for libdl')
+
+option(
+    'editline_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-leditline'],
+    description : 'link args for editline_link_args')
+
+option(
+    'lowdown_link_args', 
+    type : 'array', 
+    value : [
+        '-llowdown'], 
+    description : 'link args for lowdown')
+
+option(
+    'lzma_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-llzma'],
+    description : 'link args for liblzma')
+
+option(
+    'openssl_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lcrypto'],
+    description : 'link args for openssl')
+
+option(
+    'pthread_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-pthread'],
+    description : 'link args for pthread')
+
+option(
+    'sodium_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lsodium'],
+    description : 'link args for libsodium')
+
+option(
+    'sqlite3_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lsqlite3'],
+    description : 'link args for sqlite3')
+
+option(
+    'gc_link_args',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-lgc'],
+    description : 'link args for boehm garbage collector')
+
+option(
+    'zlib_link_args', 
+    type : 'array', 
+    value : [
+        '-lz'], 
+    description : 'link args for zlib')
```

Comment:
you're really fighting the way Meson does dependencies with this setup, Meson really want's use to just use `dependency()` or `find_library()` for handling dependencies. As it is now you're going to end up fighting Meson all the way down, and you're not going to be portable to a compiler that doesn't use `-l` or `-L`, we don't even have to talk about MSVC and windows here, imagine a user adding `['-static', '-lz'`] here, clang and gcc would be fine, but rustc would choke it wants that as `['-l', 'static=z']`

Reply:
yea, this style is pretty outdated now. Originally, i had to do it to fix building on *BSD